### PR TITLE
support read endpoint by OCO_OLLAMA_API_URL from env when using ollama

### PR DESCRIPTION
**Pull request by xtliu97**
support read OCO_OLLAMA_API_URL from env to OllamaAiEngine and fix bug in getting ollma model name when slash exists

### CODE REVIEW

File: src/utils/engine.ts [hunk 1]
```diff
@@ -13,9 +13,11 @@ export function getEngine(): AiEngine {
   
   if (provider?.startsWith('ollama')) {
     const ollamaAi = new OllamaAi();
-    const model = provider.split('/')[1];
-    if (model) ollamaAi.setModel(model);
-    
+    const model = provider.replace('ollama/', '');
```
Comment (di-sukharev) [inline src/utils/engine.ts:16]: @xtliu97 wont the change from `split` to `replace` affect anything else?

Comment (xtliu97) [reply src/utils/engine.ts]: I've reviewed the code and realized that using replace could introduce issues. I’ve corrected it by using substring to ensure everything works correctly. Thanks for catching that!

**Comment by di-sukharev** [comment]
what do you expect the ollama url to be? could you provide an example please

**Comment by xtliu97** [comment]
> what do you expect the ollama url to be? could you provide an example please

I'd like to manually set the Ollama backend URL to point to a custom-deployed server other that localhost, such as one hosted on Azure or AMS.

**Comment by di-sukharev** [comment]
why do you want that? i think i understand, just doing a short customer development session here :)

**Comment by di-sukharev** [comment]
running tests

**Comment by xtliu97** [comment]
> why do you want that? i think i understand, just doing a short customer development session here :)

Sure. I'm working with a small group, and by linking to a private GPU server, I can avoid sending code externally, ensuring data compliance without running ollama server on each computer.

**Comment by di-sukharev** [comment]
that's what i though. thank you @xtliu97 

would it help if we introduce no-logs stateless absolutely private gpu server that runs a powerful llama model for you? you'd pay per token and save money on running your own server when it's stale

**Review by di-sukharev — APPROVED** [review]

**Comment by xtliu97** [comment]
> that's what i though. thank you @xtliu97 
> 
> would it help if we introduce no-logs stateless absolutely private gpu server that runs a powerful llama model for you? you'd pay per token and save money on running your own server when it's stale. it would be a no-logs shared instance running on AWS where nobody can see a bit of any of your requests

That sounds great. Using a private service and paying by token effectively addresses local server idling and code privacy concerns.

**Comment by di-sukharev** [comment]
then we are working on that ;)

**Comment by di-sukharev** [comment]
@xtliu97 do you know that openai and anthropic retains data for 30 days only? does it steal breach data compliance you expect to have?

**Pull request merged**